### PR TITLE
[5.2] SEF: Add "Strict Routing" option

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
Joomla has been improving its SEO performance constantly and with hopefully 5.2 we will allow for a stricter routing, which removes a lot of duplicate content. This PR introduces a switch to enable all these new behaviors, but does not implement them itself. It is basically a preparation for #42850, #42854 and #42989.

### Testing Instructions
Codereview

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by robbiejackson** [comment]
I've tested this functionality on my Joomla 5.1 instance and it worked ok.

I checked that the form displayed correctly, showed the correct online help and stored the correct values in the params field in the database.

**Comment by Quy** [comment]
Please go to the issue tracker here https://issues.joomla.org/tracker/joomla-cms/43432 and use the blue "Test this" button at the top left corner, select your test result and submit. Otherwise your test will not be counted. A PR needs 2 successful human tests to get accepted.

**Comment by robbiejackson** [comment]
I have tested this item :white_check_mark: successfully on aaf16845b2ca05f5c13f65bc53b4525d0e0d0876<br /><br />I checked that the form displayed correctly, showed the correct online help and stored the correct values in the params field in the database.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43432">issues.joomla.org/tracker/joomla-cms/43432</a>.</sub>

File: administrator/language/en-GB/plg_system_sef.ini [hunk 1]
```diff
@@ -7,6 +7,8 @@ PLG_SEF_DOMAIN_DESCRIPTION="If your site can be accessed through more than one d
 PLG_SEF_DOMAIN_LABEL="Site Domain"
 PLG_SEF_INDEXPHP_DESCRIPTION="This option enables a stricter handling of 'index.php' in URLs when 'Use URL Rewriting' is enabled in Global Configuration. It will remove 'index.php' if a URL still contains it and redirect incoming requests with 'index.php' to the version without the 'index.php'."
 PLG_SEF_INDEXPHP_LABEL="Strict handling of index.php"
+PLG_SEF_STRICTROUTING_DESCRIPTION="Strict Routing will prevent the router from accepting a lot of URLs which are deemed as duplicates and will automatically redirect to the correct URL with a 301."
```
Comment (Quy) [inline administrator/language/en-GB/plg_system_sef.ini]:
```suggestion
PLG_SEF_STRICTROUTING_DESCRIPTION="Prevent the router from accepting a lot of URLs which are deemed as duplicates and will automatically redirect to the correct URL with a 301."
```

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on cc0c9194524c41069147cbca6090aac20a16ca65<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43432">issues.joomla.org/tracker/joomla-cms/43432</a>.</sub>

**Comment by SniperSister** [comment]
I have tested this item :white_check_mark: successfully on cc0c9194524c41069147cbca6090aac20a16ca65<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43432">issues.joomla.org/tracker/joomla-cms/43432</a>.</sub>

**Comment by pe7er** [comment]
Thanks @Hackwar !

**Comment by SemaphoreOxalis** [comment]
Hi,
Before J!5 I red that any new feature could be validated without documentation. So could we have documentation about this one because it seems a nice one but I don't get why, how and what is the impact (good or not) of each option.
Thanks for the work !

**Pull request merged**